analyzer/codechecker_analyzer/cmd/parse.py

```diff
@@ -649,6 +651,35 @@ def parse_convert_reports(input_dirs: List[str],
             number_of_reports
 
 
+def _generate_json_output(
+        severity_map: Dict, input_dirs: List[str], output_type: str,
+        output_path: str, trim_path_prefixes: Optional[List[str]],
+        skip_handler: Callable[[str], bool]
+) -> int:
+    try:
+        reports, number_of_reports = _parse_convert_reports(
+            input_dirs, output_type, severity_map, trim_path_prefixes,
+            skip_handler)
+        output_text = json.dumps(reports)
+
+        if output_path:
+            output_path = os.path.abspath(output_path)
+
+            if not os.path.exists(output_path):
+                os.mkdir(output_path)
+
+            output_file_path = os.path.join(output_path, 'reports.json')
+            with open(output_file_path, mode='w', encoding='utf-8',
+                      errors="ignore") as output_f:
+                output_f.write(output_text)
+
+        print(output_text)
+        sys.exit(2 if number_of_reports else 0)
```

Comment:
Where you call this function below you will exit based on the return value. This function should return only an exit code:
```suggestion
        return 2 if number_of_reports else 0
```

Reply:
Done.